index.js

```diff
@@ -1,3 +1,95 @@
-// javascript
+// Capture user's input on form submission
+let form = document.querySelector("form");
 
-alert("Hello from index.js!");
+form.addEventListener("submit", function (event) {
+  event.preventDefault();
+
+  const maxHoursPerLevel = {
+    basic: 5,
+    advanced: 10,
+  };
+
+  // Store the user's email address as userEmail (string/text)
+  let userEmail = document.querySelector("#email").value;
+  // Store the user's level as userLevel (string/text)
+  let userLevel = document.querySelector("#level").value;
+  // Store the user's hours of study as userHours (number)
+  let userHours = document.querySelector("#hoursPerWeek").value;
+
+  // Validate the user's input
+  let errors = {};
+
+  // Helper function to add error messages
+  function addError(field, message) {
+    if (!errors[field]) {
+      errors[field] = { messages: [] };
+    }
+    errors[field].messages.push(message);
+  }
+
+  // Check if the user has provided an email address
+  if (userEmail === "") {
+    addError("email", "Please enter your email address.");
+  }
+
+  // Check if the user has selected a level
+  if (userLevel === "") {
+    addError("level", "Please select a level of study");
+  }
+
+  // Check if the user has specified at least one hour of study
+  if (isNaN(userHours) || userHours < 1) {
+    addError("hoursPerWeek", "Please enter at least one hour of tuition.");
+  }
+
+  // Check if the userLevel exists in the maxHoursPerLevel object
+  if (!maxHoursPerLevel.hasOwnProperty(userLevel)) {
+    addError("level", "Invalid level of study selected.");
+  }
+
+  // Check if the number of hours requested is within the allowed range
+  const maxAllowedHours = maxHoursPerLevel[userLevel];
+  if (userHours > maxAllowedHours) {
+    addError(
+      "hoursPerWeek",
+      `You can only study a maximum of ${maxAllowedHours} hours per week.`
+    );
+  }
+
+  // Add error class to input elements with errors
+  for (let field in errors) {
+    let inputElement = document.querySelector(`#${field}`);
+    let labelElement = document.querySelector(`label[for=${field}]`);
+    if (inputElement) {
+      inputElement.classList.add("error-input");
+    }
+    if (labelElement) {
+      labelElement.classList.add("error-label");
+    }
+
+    // Populate the error message div with an unordered list of error messages
+    let errorDiv = document.querySelector(`#${field}-error`);
+    if (errorDiv) {
+      errorDiv.classList.add("error-message");
+      let ul = document.createElement("ul");
+
+      errors[field].messages.forEach((message) => {
+        let li = document.createElement("li");
+        li.textContent = message;
+        ul.appendChild(li);
+      });
+
+      errorDiv.innerHTML = ""; // Clear any existing content
+      errorDiv.appendChild(ul);
+    }
+ }
+
+    // Return early if there are any errors
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+  console.log({ userEmail, userLevel, userHours });
+  console.log({ errors });
```

Comment:
Excellent work. You are displaying error messages successfully to the user.

It looks as though you are ready to start the tutorial on [refactoring your code](https://gedsargent.github.io/code-monster/guides/programming/harmony-music/code-refactor/).